migrations/versions/0320_optimise_notifications.py

```diff
@@ -0,0 +1,79 @@
+"""
+
+Revision ID: 0320_optimise_notifications
+Revises: 0319_contact_list_archived
+Create Date: 2020-03-26 11:16:12.389524
+
+"""
+import os
+
+from alembic import op
+
+revision = '0320_optimise_notifications'
+down_revision = '0319_contact_list_archived'
+environment = os.environ['NOTIFY_ENVIRONMENT']
+
+
+def upgrade():
+    # We like to run this operation on live via the command prompt, to watch the progress and stop if necessary
+    if environment not in ["live", "production"]:
+        # Drop indexes notifications - no need to recreate these indexes
+        op.execute('DROP INDEX IF EXISTS ix_notifications_key_type')
+        op.execute('DROP INDEX IF EXISTS ix_notifications_api_key_id')
+        op.execute('DROP INDEX IF EXISTS ix_notifications_notification_status')
+        op.execute('DROP INDEX IF EXISTS ix_notifications_notification_type')
+        op.execute('DROP INDEX IF EXISTS ix_notifications_service_id')
+
+        # Create new composite indexes
+        # PLEASE NOTE: that if you create index on production you need to add concurrently to the create statement,
+        # however we are unable to do that inside a transaction like this upgrade method
+        ix_notifications_service_id_composite = """
+        CREATE INDEX ix_notifications_service_id_composite 
+            on notifications (service_id, notification_type, notification_status, created_at)
+        """
+        op.execute(ix_notifications_service_id_composite)
+
+        ix_notifications_notification_type_composite = """
+            CREATE INDEX ix_notifications_notification_type_composite 
+            on notifications (notification_type, notification_status, created_at)
+        """
+        op.execute(ix_notifications_notification_type_composite)
+        # DROP and CREATE all other indexes
```

Comment:
why are we doing this?

Reply:
It was from a query on the list of bad performing queries... I thought it was for the platform-admin/summary page. But it's actually used for `timeout_notifications`

Reply:
That task has taken up to 15 minutes before - we think it's worth it

Reply:
no i mean, why are we dropping and recreating all the other existing indexes?

Reply:
answered via slack:

> So by dropping an index we saw a decrease in 10GB used for a single 
> we are doing the drop and create index concurrently to clean the shared buffers
> when we re-created it it only took 1GB

Also of note: REINDEX holds an exclusive table lock, while this can be recreated concurrently